File: solution/2373. Largest Local Values in a Matrix/2373.cpp
class Solution {
public:
    vector<vector<int>> largestLocal(vector<vector<int>>& grid) {
        const int n = grid.size();
        vector<vector<int>> res(n - 2, vector<int>(n - 2));
        for (int i = 0; i < n - 2; ++i) {
            for (int j = 0; j < n - 2; ++j) {
                for (int y = i; y < i + 3; ++y)
                    for (int x = j; x < j + 3; ++x)
                        res[i][j] = max(res[i][j], grid[y][x]);
            }
        }
        return res;
    }
};

File: solution/2373. Largest Local Values in a Matrix/2373.py
class Solution:
    def largestLocal(self, grid: List[List[int]]) -> List[List[int]]:
        n = len(grid)
        
        res = [[0] * (n - 2) for _ in range(n - 2)]
        for i in range(n - 2):
            for j in range(n - 2):
                for y in range(i, i + 3):
                    for x in range(j, j + 3):
                        res[i][j] = max(res[i][j], grid[y][x])
        return res
    
